### ISSUE TITLE
MRTK Standard Shader ignoring directional light when assigned at runtime on an Asset Bundle

### ISSUE THREAD

**Issue by Vangaorth** [microsoft/MixedRealityToolkit-Unity#9637]
## Describe the bug

On the Hololens2 device, when loading an Asset Bundle at runtime, if the shader of any of its materials is replaced with the MRTK Standard Shader then no directional light is applied and the rendering appears flat. In editor, the rendering is fine.

This also happens if, instead of changing the shader, a pre-arranged material with the MRTK Standard shader is cloned and assigned at runtime, replacing the source mesh material and having its properties assigned using the Unity method "CopyPropertiesFromMaterial". This problems also appears in editor

This does not happen if the pre-arranged material (with the MRTK Standard Shader) is instantiated and assgined at runtime, replacing the source mesh material and just copying the _Color and _MainText properties from the source material to the new instantiated one. This also works fine in editor

## To reproduce

Shortest: install and run this app [(Flat Shading HL2 App)](https://github.com/Vangaorth/MRTKShaderRuntime/releases/download/1.0.0.0/0_Shader_1.0.0.0_ARM64.zip)
Short: clone this repository [(Flat Shading HL2 Source)](https://github.com/Vangaorth/MRTKShaderRuntime) build and deploy on an HL2.

In the shown scene, you will see four models (from left to right):
1) The first is just the spawned Asset Bundle with the standard Unity Shader. It looks fine in editor and on the device
2) The second is the spawned Asset Bundle where every material had its shader replaced by the MRTK Standard Shader. While it looks fine in Editor, on the device there is no shading and it appears flat
3) The third is the spawned Asset Bundle where every material has been replaced by a cloned material previously created in editor. This material has the MRTK Standard shader already applied. After the replacement, the _Color and _MainTex properties were copied from the original material to the replaced one. It looks fine in Editor and on the device.
4) The fourth is as the third but instead of copying just the two _Color and _MainTex properties, all properties have been copied using the CopyPropertiesFromMaterial (replacementMaterial.CopyPropertiesFromMaterial(originalMaterial)). Both in Editor and on the device, the rendering is flat (so it does NOT look fine).

Long: 
1. Create a new Unity 2019.4.x Project
2. Import any MRTK version 2+
3. Create an Asset Bundle for a 3D model that has some depth
4. Make sure the scene has some directional light 
5. Load the Asset Bundle at runtime and instantiate it
6. Replace the shader of every material on every MeshRenderer with the MRTK Standard Shader

## Expected behavior

The directional light should be applied when the shader is replaced at runtime and is should not look flat

## Screenshots

This is how the model should look
![Capture](https://user-images.githubusercontent.com/5150343/114011042-110b3c80-9865-11eb-9363-efbad57020dc.PNG)

This is how, under the described conditions, it looks on the device
![Capture1](https://user-images.githubusercontent.com/5150343/114011160-34ce8280-9865-11eb-8e26-331cdcc01f53.PNG)

This is a comparison to make it clearer
![Capture3](https://user-images.githubusercontent.com/5150343/114011708-d5bd3d80-9865-11eb-9171-5509f542ea57.png)

## Your setup

- Unity 2019.4.23
- MRTK Version 2.+ (currently using 2.6.1)

## Target platform

- HoloLens 2 (with the latest system update)

## Additional context

Some workarounds suggested on the internet were tried with no success. Some of them are:
- Changing the "Albedo at Runtime" property on the  material
- Enabling/disabling Main Texture ( material.EnableKeyword("_MainTex"); )
- Enabling/disabling Directional Light ( material.EnableKeyword("_DIRECTIONAL_LIGHT"); ) and/or Specular Highlights ( material.EnableKeyword("_SPECULAR_HIGHLIGHTS"); )

This entire issue does not happen if the loaded Asset Bundle already has the MRTK Standard shader assigned on its materials when it was originally created (e.g., an FBX is imported in Unity, the shaders are replaced with the MRTK Standard Shader, the Asset Bundle is created in editor and later, at runtime, is loaded).

**Comment by RogPodge** [comment]
@Cameron-Micka wondering if your shader expertise might be able to inform what's going on here.

**Comment by Cameron-Micka** [comment]
Thank you for the in depth bug report @Vangaorth! Based on the description my guess is that a shader variant (needed for lighting) is not being compiled for the build. Which sounds somewhat similar to [this bug/pr](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/7543).

I'm in the middle of a MRTK-Unreal release, but once that is complete (which should be in the next few days). I'll pull down your example and take a deeper look.

**Comment by IssueSyncBot** [comment]
We appreciate your feedback and thank you for reporting this issue.

Microsoft Mixed Reality Toolkit version 2 (MRTK2) is currently in limited support. This means that Microsoft is only fixing high priority security issues. Unfortunately, this issue does not meet the necessary priority and will be closed. If you strongly feel that this issue deserves more attention, please open a new issue and explain why it is important.

Microsoft recommends that all new HoloLens 2 Unity applications use MRTK3 instead of MRTK2.

Please note that MRTK3 was released in August 2023. It features an all new architecture for developing rich mixed reality experiences and has a minimum requirement of Unity 2021.3 LTS. For more information about MRTK3, please visit[https://www.mixedrealitytoolkit.org](https://www.mixedrealitytoolkit.org).

Thank you for your continued support of the Mixed Reality Toolkit!
